### DISCUSSION TITLE
Updating a table with the table editor delete the composite foreign key constraint on the table

### DISCUSSION THREAD

**Discussion by atrahy** [supabase/supabase#16956] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

TL;DR: updating a table with the table editor delete the composite foreign key constraint on the table

More context

I create two table, the first one with a composite primary key and the second one with the composite foreign key to reference the first table.
If I edit the table on the dashboard (even if I edit nothing) and click on save, an error pop-up and the relation is deleted post update.

I am conscious that composite foreign keys are not supported by the dashboard but updating the table, without editing this constraint, should not delete it.

## To Reproduce

1. Create the two table mentioned above

- `table1` has the composite primary key

<img width="651" alt="Screenshot 2023-03-27 at 23 07 52" src="https://user-images.githubusercontent.com/24977757/227972341-3a117e5c-3c81-4406-b47c-aee9cda329d0.png">

- `table2` has the composite foreign key

<img width="643" alt="Screenshot 2023-03-27 at 23 08 44" src="https://user-images.githubusercontent.com/24977757/227972483-2a0724ec-4a72-49b4-a768-713b9a87b2cf.png">

2. Create a composite foreign key constraint with SQL editor

```sql
ALTER TABLE table2
  ADD CONSTRAINT table1_fkey FOREIGN KEY (id_table1, date_table1)
    REFERENCES table1(id, date); 
```

3. See that the constraint has been created (used `psql` cli with `\d [table_name]`)

```
                                      Table "public.table2"
   Column    |           Type           | Collation | Nullable |             Default
-------------+--------------------------+-----------+----------+----------------------------------
 id          | bigint                   |           | not null | generated by default as identity
 created_at  | timestamp with time zone |           |          | now()
 id_table1   | bigint                   |           |          |
 date_table1 | date                     |           |          |
Indexes:
    "table2_pkey" PRIMARY KEY, btree (id)
Foreign-key constraints:
    "table1_fkey" FOREIGN KEY (id_table1, date_table1) REFERENCES table1(id, date)
Policies (row security enabled): (none)
```

3. Update the table on the dashboard
    - Go to Table edit view
    - Click on drop down arrow next to the table name
    - Click edit table
    - Add a new column (here a column `name`) or do nothing
    - Click "Save"
4. See errors
Theses errors seams to imply that the dashboard tried to re-create the foreign key constraints but since the target column does not have a `UNIQUE` constraint, the error pops-up and the foreign key is just skipped. If the table match the `UNIQUE` constraints 
<img width="337" alt="Screenshot 2023-03-27 at 23 12 49" src="https://user-images.githubusercontent.com/24977757/227973435-f0c06352-fd77-4cb7-ae12-d722680efa5e.png">

5. See the constraint is not available

```
                                      Table "public.table2"
   Column    |           Type           | Collation | Nullable |             Default
-------------+--------------------------+-----------+----------+----------------------------------
 id          | bigint                   |           | not null | generated by default as identity
 created_at  | timestamp with time zone |           |          | now()
 id_table1   | bigint                   |           |          |
 date_table1 | date                     |           |          |
 name        | text                     |           |          |
Indexes:
    "table2_pkey" PRIMARY KEY, btree (id)
Policies (row security enabled): (none)
```

## Expected behavior

The update of the table (name / description) or other column than the one included in the composite foreign key should not affect existing constraints.
For exemple, only adding or updating a column with the designated button (Database -> Tables -> click on "X Columns" -> Add column OR Edit button of one column) does not trigger the error. 

The best would be a support of composite foreign keys but that's not the point here.

## System information

- OS: macOs
- Browser: chrome

**Comment by MildTomato** [comment]
Hey folks - me and the team here are looking at a visual solution in the dashboard for composite foreign keys today and I'll post an update shortly on what we might propose.

**Comment by MildTomato** [comment]
related issue https://github.com/supabase/supabase/issues/3583#issuecomment-1552370900

**Comment by MildTomato** [comment]
Some updates [on the other issue](https://github.com/supabase/supabase/issues/3583#issuecomment-1553952927)
